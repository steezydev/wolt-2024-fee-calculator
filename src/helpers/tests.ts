const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
