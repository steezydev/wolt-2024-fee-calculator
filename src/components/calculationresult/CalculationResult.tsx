import B1 from '@/components/typography/B1';
import { classNames } from '@/helpers/classNames';
import { formatCurrency } from '@/helpers/currency';
import { CalculationResultProps } from '@/types/props/CalculationResultProps';

const CalculationResult = ({
  result,
  currency = 'EUR',
  children,
  className,
}: CalculationResultProps) => {
  return (
    <div
      className={classNames(
        'flex flex-col gap-6 bg-primary-100 py-6 px-5 rounded-xl text-black dark:text-white',
        className
      )}
    >
      <div className='flex flex-col gap-1'>{children}</div>
      <div className='flex flex-row justify-between'>
        <B1 className='text-2xl font-semibold'>Delivery fee</B1>
        <B1 className='text-2xl  font-semibold'>
          {formatCurrency(result, currency)}
        </B1>
      </div>
    </div>
  );
};

export default CalculationResult;
