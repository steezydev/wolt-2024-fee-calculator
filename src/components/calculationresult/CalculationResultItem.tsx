import B3 from '@/components/typography/B3';
import { classNames } from '@/helpers/classNames';
import { CalculationResultItemProps } from '@/types/props/CalculationResultItemProps';

const CalculationResultItem = ({
  label,
  value,
  className,
}: CalculationResultItemProps) => {
  return (
    <div className={classNames('flex flex-row justify-between', className)}>
      <B3>{label}</B3>
      <B3>{value}</B3>
    </div>
  );
};

export default CalculationResultItem;
