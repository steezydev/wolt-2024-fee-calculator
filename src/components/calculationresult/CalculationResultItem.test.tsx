import { calculationResultItemRenderTests } from './CalculationResult.tests';
import CalculationResultItem from './CalculationResultItem';

calculationResultItemRenderTests(
  'CalculationResultItem',
  CalculationResultItem
);
