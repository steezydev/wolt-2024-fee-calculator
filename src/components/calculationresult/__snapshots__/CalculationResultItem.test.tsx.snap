// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculationResultItem when asked to render renders 1`] = `
<div
  className="flex flex-row justify-between"
>
  <p
    className="text-base"
  >
    Cart value
  </p>
  <p
    className="text-base"
  >
    20 €
  </p>
</div>
`;

exports[`CalculationResultItem when asked to render with additional class renders with additional class 1`] = `
<div
  className="flex flex-row justify-between some-class"
>
  <p
    className="text-base"
  >
    Cart value
  </p>
  <p
    className="text-base"
  >
    20 €
  </p>
</div>
`;
